plugin/http/httptrace/httptrace.go

```diff
@@ -59,10 +62,38 @@ func (t *Transport) RoundTrip(req *http.Request) (*http.Response, error) {
 	resp, err := t.base().RoundTrip(req)
 
 	// TODO(jbd): Add status and attributes.
-	trace.EndSpan(ctx)
+	if err != nil {
+		trace.EndSpan(ctx)
+		return resp, err
+	}
+
+	// trace.EndSpan(ctx) will be invoked after
+	// resp.Body.Close() has been invoked.
+	resp.Body = &bodyProxyReader{rc: resp.Body, ctxToEnd: ctx}
 	return resp, err
 }
 
+type bodyProxyReader struct {
```

Comment:
proxy is maybe a bit misleading here (might be interpreted as HTTP proxy), what about something like tracedBody?